plugin/plugin.go

```diff
@@ -171,30 +174,41 @@ func run(ctx context.Context, pluginName string, pluginPath string, req proto.Re
 	// serialize request
 	data, err := json.Marshal(req)
 	if err != nil {
-		return fmt.Errorf("%s: failed to marshal request object: %w", pluginName, err)
+		logger.Errorf("failed to marshal request: %+v", req)
+		return PluginUnknownError{
+			Msg:        fmt.Sprintf("failed to execute the %s command for %s plugin", req.Command(), pluginName),
```

Comment:
`failed to marshal request` is acceptable for developers, but for users, it may be confusing because they might not be familiar with the Notation plugin protocol and don't know what is happening. When the CLI receives a `PluginUnknownError`, it generates a recommendation suggesting the user contact the Notation developer. Therefore, simply telling the user that the execution failed is sufficient, and the user should provide a debug log for developers to conduct further investigation.

Reply:
```go
		return PluginUnknownError{
			Msg:        fmt.Sprintf("failed to execute the %s command for %s plugin", req.Command(), pluginName),
			InnerError: fmt.Errorf("failed to marshal request: %w", err)}
```
Updated to return the `failed to marshal request` in InnerError` for notation-go user.